### PR TITLE
audit improvements

### PR DESCRIPTION
**Pull request by lemingos**
Improvements to MK audit 

- [x] Show only warnings for tags/filters that really exists in the repo
- [x] Search through all liquid files including forms
- [x] Color adjustments see screenshot below
- [x] New rules for recently deprecated features custom_attributes, snake_case model names, attribute_type, configuration etc.
- [x] Formatting improvements, changes in messages

```
audit_test % mk audit

[Audit Warn] deprectaed "title" tag in the following files:
  marketplace_builder/views/pages/home.html.liquid
  marketplace_builder/views/partials/layout/constants.liquid
  marketplace_builder/views/partials/layout/head.liquid


[Audit Warn] deprecated "generate_url" filter in the following files:
  marketplace_builder/form_configurations/test.liquid


[Audit Warn] deprecated argument "resource_id" of include_form tag, please replace with "id" in the following files:
  marketplace_builder/views/pages/home.html.liquid


grep: no such file or directory: {marketplace_builder,modules/**}/{views/**/*.liquid
[Audit Warn] deprecated key "configuration", please replace with "fields" in the following files:
  marketplace_builder/form_configurations/test.liquid


[Audit Warn] deprecated key "attribute_type", please replace with "type" in the following files:
  marketplace_builder/custom_model_types/test.yml
  marketplace_builder/custom_model_types/valid_test.yml


[Audit Warn] please use snake_case format as model name in the following files:
  marketplace_builder/custom_model_types/test.yml


[Audit Warn] deprecated key "custom_attributes", please replace with "properties" in the following files:
  marketplace_builder/custom_model_types/test.yml
  marketplace_builder/custom_model_types/valid_test.yml
```

Possible future improvements:
- deprecated tags, filters library should be array of objects not just names similar structure to the base rules so that proper regex can be applied currently tag title maches all title occurences in the views
- add unit test coverage with Jasmine

### CODE REVIEW

**Comment by pavelloz** [comment]
In the message you pasted above there is this:

```grep: no such file or directory: {marketplace_builder,modules/**}/{views/**/*.liquid
[Audit Warn] deprecated key "configuration", please replace with "fields" in the following files:
  marketplace_builder/form_configurations/test.liquid
```

I assume its typo in `{views`

File: lib/audit/rules.js [hunk 1]
```diff
@@ -19,31 +27,57 @@ const log = {
...
   {
-    directory: 'views/**/*.liquid',
-    find: new RegExp(DEPRECATED_FILTERS.join('|') + ' |'),
-    message: () => {
-      return log.warning(
-        `You are using deprecated liquid filters.
-        \nLook for those in your code: ${DEPRECATED_FILTERS.join(', ')}`
-      );
-    }
+    directory: '{views/**/*.liquid',
```
Comment (pavelloz) [inline lib/audit/rules.js]:
```suggestion
    directory: '{views}/**/*.liquid',
```

**Comment by pavelloz** [comment]
One suggestion, after that all good

File: lib/audit/rules.js [hunk 2]
```diff
@@ -19,31 +27,57 @@ const log = {
...
-
-module.exports = [
+const baseRules = [
   {
     directory: 'views/pages/**/*.liquid',
     find: /^enable_profiler\: true$/,
-    message: matches => {
-      log.warning(`Found "enable_profiler: true" setting in ${matches.length} files. This will result in a performance hit.`);
-    }
+    message: printMessage('plese remove "enable_profiler: true" setting')
```
Comment (pavelloz) [inline lib/audit/rules.js]:
```suggestion
    message: printMessage('please remove "enable_profiler: true" setting')
```

File: lib/audit/rules.js [hunk 3]
```diff
@@ -19,31 +27,57 @@ const log = {
...
+
+const deprecatedTagRules = DEPRECATED_TAGS.map( function(val) {
+  return processRules(val, `deprectaed "${val}" tag` );
+});
+
+const deprecatedFilterRules = DEPRECATED_FILTERS.map( function(val) {
+  return processRules(val, `deprectaed "${val}" filter`);
+});
+
+module.exports = [ ...deprecatedTagRules, ...deprecatedFilterRules, ...baseRules];
```
Comment (pavelloz) [inline lib/audit/rules.js]:
```suggestion
module.exports = [ ...deprecatedTagRules, ...deprecatedFilterRules, ...baseRules ];
```

File: lib/audit/rules.js [hunk 4]
```diff
@@ -19,31 +27,57 @@ const log = {
...
+const processRules = (val, message) => {
+  return {
+    directory: './**/*.liquid',
+    find: new RegExp(val),
+    message: printMessage(message)
+  };
+};
+
+const deprecatedTagRules = DEPRECATED_TAGS.map( function(val) {
+  return processRules(val, `deprectaed "${val}" tag` );
```
Comment (pavelloz) [inline lib/audit/rules.js]:
```suggestion
  return processRules(val, `deprected "${val}" tag` );
```

**Comment by lemingos** [comment]
ping @pavelloz

**Pull request merged**